examples/simple/hamming.c

```diff
@@ -0,0 +1,21 @@
+#include <igraph.h>
+#include <stdio.h>
+
+int main() {
+    igraph_t g;
+    igraph_error_t ret;
+
+    // Build Hamming graph H(3,2) → 8 vertices, 12 edges
+    ret = igraph_hamming_graph(&g, 3, 2);
+    if (ret) {
+        fprintf(stderr, "igraph_hamming_graph failed!\n");
+        return 1;
+    }
+
+    printf("H(3,2):\n");
+    printf("  Vertices: %ld\n", (long) igraph_vcount(&g));
+    printf("  Edges: %ld\n", (long) igraph_ecount(&g));
```

Comment:
To follow conventions, never use `long` in igraph code, not even in tests.

Ideally, use `IGRAPH_PRId` instead of `%ld`. Alternatively, if you know the value fits in 32 bits, you can cast to `int` and use `%d`.